mongodb/src/main/java/org/hibernate/ogm/datastore/mongodb/impl/MongoDBIndexesDefiner.java

```diff
@@ -0,0 +1,270 @@
+			UniqueConstraintSchemaUpdateStrategy constraintMethod) {
+		Iterator<UniqueKey> keys = table.getUniqueKeyIterator();
+		while ( keys.hasNext() ) {
+			UniqueKey uniqueKey = keys.next();
+			forIndexNotReferenced.remove( uniqueKey.getName() );
+
+			if ( constraintMethod != UniqueConstraintSchemaUpdateStrategy.SKIP ) {
+				MongoDBIndexSpec indexSpec = new MongoDBIndexSpec( uniqueKey,
+						getIndexOptionDocument( table, indexOptions.getOptionForIndex( uniqueKey.getName() ) ) );
+				if ( validateIndexSpec( indexSpec ) ) {
+					addHandleDuplications( indexSpec );
+				}
+			}
+		}
+	}
+
+	void validateIndexSpecsForIndexes(Table table, IndexOptions indexOptions, Set<String> forIndexNotReferenced) {
+		Iterator<Index> indexes = table.getIndexIterator();
+		while ( indexes.hasNext() ) {
+			Index index = indexes.next();
+			forIndexNotReferenced.remove( index.getName() );
+			MongoDBIndexSpec indexSpec = new MongoDBIndexSpec( index,
+					getIndexOptionDocument( table, indexOptions.getOptionForIndex( index.getName() ) ) );
+			if ( validateIndexSpec( indexSpec ) ) {
+				addHandleDuplications( indexSpec );
+			}
+		}
+	}
+
+	private void addHandleDuplications(MongoDBIndexSpec indexSpec) {
+		// we can add any index defined on different collection or different key set
+		// we rely on hashcode and equals to verify conflicts
+		if ( !indexSpecs.contains( indexSpec ) ) {
+			indexSpecs.add( indexSpec );
+			return;
+		}
+
+		int existingIndexPosition = indexSpecs.indexOf( indexSpec );
+		MongoDBIndexSpec existingIndex = indexSpecs.get( existingIndexPosition );
+
+		// in case on conflict
+		// we rely on compareTo to choose the best candidate
+		if ( indexSpec.compareTo( existingIndex ) < 0 ) {
+			// replace the existing definition with the new one
+			indexSpecs.remove( existingIndexPosition );
+			indexSpecs.add( indexSpec );
+			log.tryToDefineMultipleIndexesOnTheSameKeySet( indexSpec.getIndexKeysDocument().toJson(), existingIndex.getIndexName() );
+		}
+		else {
+			// discard the current definition
+			log.tryToDefineMultipleIndexesOnTheSameKeySet( indexSpec.getIndexKeysDocument().toJson(), indexSpec.getIndexName() );
+		}
+	}
+
+	private Document getIndexOptionDocument(Table table, IndexOption indexOption) {
+		try {
+			Document options;
+			if ( StringHelper.isNullOrEmptyString( indexOption.getOptions() ) ) {
+				options = new Document();
+			}
+			else {
+				options = Document.parse( indexOption.getOptions() );
+			}
+			options.put( "name", indexOption.getTargetIndexName() );
+			return options;
+		}
+		catch (Exception e) {
+			throw log.invalidOptionsFormatForIndex( table.getName(), indexOption.getTargetIndexName(), e );
+		}
+	}
+
+	private boolean validateIndexSpec(MongoDBIndexSpec indexSpec) {
+		boolean valid = true;
+		if ( StringHelper.isNullOrEmptyString( indexSpec.getIndexName() ) ) {
+			log.indexNameIsEmpty( indexSpec.getCollection() );
+			valid = false;
```

Comment:
can't you just return false here and avod the use of variable?

Reply:
Never mind, the code was already like this

Reply:
Yes unfortunately the move of the method from another class it is not visible.